airflow-core/src/airflow/ui/tests/e2e/pages/HomePage.ts

```diff
@@ -0,0 +1,181 @@
+    this.metaDatabaseHealth = page.getByText("Metadatabase").first();
+    this.schedulerHealth = page.getByText("Scheduler").first();
+    this.triggererHealth = page.getByText("Triggerer").first();
+    this.dagProcessorHealth = page.getByText("DAG Processor").first();
+
+    // Pool Summary section
+    this.poolSummarySection = page.locator('text="Pool Summary"').locator("..").locator("..");
+
+    // Historical Metrics section (recent runs)
+    this.historicalMetricsSection = page.getByText("History").first().locator("..").locator("..");
+    this.dagRunMetrics = page.getByRole("heading", { name: /dag run/i }).first();
+    this.taskInstanceMetrics = page.getByRole("heading", { name: /task instance/i }).first();
+  }
+
+  /**
+   * Check if health badges are visible
+   */
+  public async areHealthBadgesVisible(): Promise<boolean> {
+    try {
+      await this.metaDatabaseHealth.waitFor({ state: "visible", timeout: 10_000 });
+      await this.schedulerHealth.waitFor({ state: "visible", timeout: 5000 });
+
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Get Active DAGs count
+   */
+  public async getActiveDagsCount(): Promise<number> {
+    return this.getStatsCardCount(this.activeDagsCard);
+  }
+
+  /**
+   * Get Failed DAGs count
+   */
+  public async getFailedDagsCount(): Promise<number> {
+    return this.getStatsCardCount(this.failedDagsCard);
+  }
+
+  /**
+   * Get Running DAGs count
+   */
+  public async getRunningDagsCount(): Promise<number> {
+    return this.getStatsCardCount(this.runningDagsCard);
+  }
+
+  /**
+   * Check if DAG Import Errors are displayed (only visible when errors exist)
+   */
+  public async isDagImportErrorsVisible(): Promise<boolean> {
+    try {
+      return await this.dagImportErrorsCard.isVisible();
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Check if historical metrics (recent runs) section is visible
+   */
+  public async isHistoricalMetricsSectionVisible(): Promise<boolean> {
+    try {
+      await this.dagRunMetrics.waitFor({ state: "visible", timeout: 10_000 });
+
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Check if stats section is visible
+   */
+  public async isStatsSectionVisible(): Promise<boolean> {
+    try {
+      await this.activeDagsCard.waitFor({ state: "visible", timeout: 10_000 });
+
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Navigate to Home/Dashboard page
+   */
+  public async navigate(): Promise<void> {
+    await this.navigateTo(HomePage.homeUrl);
+  }
+
+  /**
+   * Wait for dashboard to fully load
+   */
+  public async waitForDashboardLoad(): Promise<void> {
+    await this.welcomeHeading.waitFor({ state: "visible", timeout: 30_000 });
+  }
+
+  /**
+   * Get the count from a stats card
+   */
+  // eslint-disable-next-line @typescript-eslint/class-methods-use-this
+  private async getStatsCardCount(card: Locator): Promise<number> {
```

Comment:
If the card doesn't exist, it returns 0, and expect(count).toBeGreaterThanOrEqual(0) still passes. The test never verifies the element actually exists.
```suggestion
  private async getStatsCardCount(card: Locator): Promise<number> {
  await card.waitFor({ state: "visible" }); // Fail fast if card doesn't exist
  const badgeText = await card.locator("span").first().textContent();
  const match = badgeText?.match(/\d+/);
  
  if (!match) {
    throw new Error("Could not find count in stats card");
  }
  
  return parseInt(match[0], 10);
}
```

Reply:
addressed it